From kay.sievers@vrfy.org Fri May  2 09:28:41 2008
From: Greg Kroah-Hartman <gregkh@suse.de>
Date: Fri, 02 May 2008 06:02:41 +0200
Subject: driver core: remove DEVICE_ID_SIZE define
Message-ID: <1209700961.3100.5.camel@linux.site>


There is no such thing as a "device id size" in the driver core, so
remove the define and fix up any users of this odd define in the rest of
the kernel.

Cc: Kay Sievers <kay.sievers@vrfy.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/memstick/host/jmb38x_ms.c |    4 ++--
 include/linux/device.h            |    1 -
 2 files changed, 2 insertions(+), 3 deletions(-)

--- a/drivers/memstick/host/jmb38x_ms.c
+++ b/drivers/memstick/host/jmb38x_ms.c
@@ -51,7 +51,7 @@ struct jmb38x_ms_host {
 	void __iomem            *addr;
 	spinlock_t              lock;
 	int                     id;
-	char                    host_id[DEVICE_ID_SIZE];
+	char                    host_id[32];
 	int                     irq;
 	unsigned int            block_pos;
 	unsigned long           timeout_jiffies;
@@ -781,7 +781,7 @@ static struct memstick_host *jmb38x_ms_a
 
 	spin_lock_init(&host->lock);
 	host->id = cnt;
-	snprintf(host->host_id, DEVICE_ID_SIZE, DRIVER_NAME ":slot%d",
+	snprintf(host->host_id, sizeof(host->host_id), DRIVER_NAME ":slot%d",
 		 host->id);
 	host->irq = jm->pdev->irq;
 	host->timeout_jiffies = msecs_to_jiffies(1000);
--- a/include/linux/device.h
+++ b/include/linux/device.h
@@ -24,7 +24,6 @@
 #include <asm/atomic.h>
 #include <asm/device.h>
 
-#define DEVICE_ID_SIZE		32
 #define BUS_ID_SIZE		20
 
 struct device;
